plantclef/workflows/baseline.py

```diff
@@ -0,0 +1,158 @@
+from argparse import ArgumentParser
+
+import luigi
+import luigi.contrib.gcs
+from pyspark.ml import Pipeline, PipelineModel
+from pyspark.ml.classification import LogisticRegression
+from pyspark.ml.feature import SQLTransformer
+from pyspark.ml.functions import array_to_vector, vector_to_array
+from pyspark.sql import DataFrame
+from pyspark.sql import functions as F
+
+from plantclef.transforms import DCTN, WrappedDinoV2
+from plantclef.utils import spark_resource
+
+
+class ProcessBase(luigi.Task):
+    input_path = luigi.Parameter()
+    output_path = luigi.Parameter()
+    num_partitions = luigi.IntParameter(default=500)
+
+    def output(self):
+        # save both the model pipeline and the dataset
+        return luigi.contrib.gcs.GCSTarget(f"{self.output_path}/_SUCCESS")
+
+    @property
+    def feature_columns(self) -> list:
+        raise NotImplementedError()
+
+    def pipeline(self) -> Pipeline:
+        raise NotImplementedError()
+
+    def transform(self, model, df, features) -> DataFrame:
+        transformed = model.transform(df)
+        for c in features:
+            # check if the feature is a vector and convert it to an array
+            if "array" in transformed.schema[c].simpleString():
+                continue
+            transformed = transformed.withColumn(c, vector_to_array(F.col(c)))
+        return transformed
+
+    def _get_subset(self, df):
+        # Get subset of images to test pipeline
+        subset_df = (
+            df.where(F.col("species_id").isin([1361703, 1355927]))
+            .orderBy(F.rand(1000))
+            .limit(200)
+            .cache()
+        )
+        return subset_df
+
+    def run(self):
+        with spark_resource(
+            **{"spark.sql.shuffle.partitions": max(self.num_partitions, 200)}
+        ) as spark:
+            df = spark.read.parquet(self.input_path)
```

Comment:
The job might be dying because of how large the input dataset is. Can you increase this to 500 or some large number and try the process again? Maybe set it to `self.num_partitions`